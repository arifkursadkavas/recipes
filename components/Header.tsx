import * as React from 'react';
import { Component } from 'react';

class Header extends Component {
  render() {
    return (
      <div className='header'>
        <img className='headerLogo' src='favicon-96x96.png'></img>
        <span className='headerTitle'>Marley Spoon</span>
      </div>
    );
  }
}

export default Header;
