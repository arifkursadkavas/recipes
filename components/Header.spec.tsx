import * as React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Header from './Header';

describe('Components', () => {
  describe('Header', () => {
    test('snapshot renders', () => {
      const component = renderer.create(<Header />);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});

describe('Components', () => {
  describe('Header', () => {
    it('should render header text without throwing an error', function () {
      const wrap = mount(<Header />);
      expect(wrap.find('span').text()).toBe('Marley Spoon');
    });
  });
});

describe('Components', () => {
  describe('Header', () => {
    it('should render logo image with src correctly', () => {
      const wrap = mount(<Header />);
      expect(wrap.find('img').prop('src')).toEqual('../favicon-96x96.png');
    });
  });
});
