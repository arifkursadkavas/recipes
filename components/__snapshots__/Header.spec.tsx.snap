// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Header snapshot renders 1`] = `
<div
  className="header"
>
  <img
    alt="Header Logo"
    className="headerLogo"
    onClick={[Function]}
    onMouseEnter={[Function]}
    src="../favicon-96x96.png"
  />
  <span
    className="headerTitle"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Marley Spoon
  </span>
</div>
`;
